docs/src/reference.md

```diff
@@ -23,3 +23,11 @@
 Modules = [Arrow]
 Order   = [:type, :function]
 ```
+
+## Internals: `Arrow.FlatBuffers`
```

Comment:
documenter wants all docstrings to appear in the docs somewhere. So I've added a section here, but explicitly marked it as part of the internals to not change the API surface.